aca_required_temp.py

```diff
@@ -0,0 +1,189 @@
+    return stars_in_fov
+
+
+def max_temp(ra, dec, roll, time, cone_stars):
+    fov_stars = select_fov_stars(ra, dec, roll, cone_stars)
+    if not len(fov_stars):
+        return None
+    # take the 8 brightest
+    stars = fov_stars[0:8]
+    id_hash = tuple(stars['AGASC_ID'])
+    if id_hash in TEMP_CACHE:
+        t_ccd, n_acq = TEMP_CACHE[id_hash]
+    else:
+        t_ccd, n_acq = t_ccd_warm_limit(date=time,
+                                        mags=stars['MAG_ACA'],
+                                        colors=stars['COLOR1'],
+                                        min_n_acq=N_ACQ_STARS,
+                                        cold_t_ccd=COLD_T_CCD,
+                                        warm_t_ccd=WARM_T_CCD)
+        TEMP_CACHE[id_hash] = (t_ccd, n_acq)
+    return t_ccd
+
+
+def get_rolldev(pitch):
+    if pitch < ROLL_TABLE['pitch'][0]:
+        return 0
+    idx = np.searchsorted(ROLL_TABLE['pitch'], pitch)
+    return ROLL_TABLE['rolldev'][idx - 1]
+
+
+def best_temp_roll(ra, dec, nom_roll, day_pitch, time, cone_stars):
+    """
+    Loop over possible roll range for this pitch and return best temp/roll
+    combination
+    """
+    best_t_ccd = None
+    best_roll = None
+    rolldev = get_rolldev(day_pitch)
+    # Quat doesn't care about the domain of roll, so I don't have to
+    # do anything tricky with 180/360 etc
+    for rolldiff in np.arange(0, rolldev, step=.5):
+        # Walk out from nom_roll instead of just through the range,
+        # and quit if we get one that is at the warm limit
+        # That will give us the closest-to-nominal best choice
+        for roll in [nom_roll - rolldiff, nom_roll + rolldiff]:
+            roll_t_ccd = max_temp(ra, dec, roll, time=time, cone_stars=cone_stars)
+            if roll_t_ccd is not None:
+                if best_t_ccd is None or roll_t_ccd > best_t_ccd:
+                    best_t_ccd = roll_t_ccd
+                    best_roll = roll
+                if best_t_ccd == WARM_T_CCD:
+                    break
+    return best_t_ccd, best_roll
+
+
+def temps_for_attitude(ra, dec, start='2014-09-01', stop='2015-12-31'):
+    # reset the caches at every new attitude
+    global TEMP_CACHE
+    TEMP_CACHE.clear()
+
+    # set the agasc lookup time to be in the middle of the cycle for
+    # proper motion correction
+    agasc_mid_time = DateTime((DateTime(start).secs + DateTime(stop).secs)
+                              / 2).date
+
+    # Get stars in this field
+    cone_stars = get_agasc_cone(ra, dec, time=agasc_mid_time)
+
+    # get a list of days
+    start = DateTime(start)
+    stop = DateTime(stop)
+    days = start + np.arange(stop - start)
+
+    temps = {}
+    # loop over them
+    for day in days.date:
+        nom_roll = nominal_roll(ra, dec, time=day)
+        day_pitch = pitch(ra, dec, time=day)
+        if day_pitch < 45 or day_pitch > 170:
+            continue
+
+        nom_roll_t_ccd = max_temp(ra, dec, nom_roll, time=day, cone_stars=cone_stars)
```

Comment:
Delegate all this logic to a single function `get_t_ccd_roll` instead of having two places where there is a decision about what is best and having two places that call `max_temp`.  This way you kill the `nom_roll = ..` call above and have a single call in this loop:
```
nom_t_ccd, nom_roll, best_t_ccd, best_roll = get_t_ccd_roll(
        ra, dec, pitch, time=day, cone_stars=cone_stars)
rows.append((day, pitch, nom_roll, nom_t_ccd, best_roll, best_t_ccd))
```
Previously you would have done `rows = []` and finish with below (with no need to sort):
```
table = Table(rows=rows, names=(...))
for name in ('pitch', ...):
    table[name].format = '{:.2f}'
```